--- bin/speedseq.orig	2015-03-30 02:11:41.000000000 +0800
+++ bin/speedseq	2015-08-06 11:41:01.115339000 +0800
@@ -635,7 +635,6 @@
             -f $REF \\
             --region \$chrom:\$start..\$end \\
             $BAM_STRING \\
-            --experimental-gls \\
             --min-repeat-entropy 1 \\
             | $VAWK --header '\$6>=$MINQUAL && I\$RPR>0 && I\$RPL>0' \\
             > ${TEMP_DIR}/$OUTBASE.\$i.vcf"
@@ -646,7 +645,6 @@
         echo "$FREEBAYES \
         -f $REF \
         --region $i \
-        --experimental-gls \
         --min-repeat-entropy 1 \
         $BAM_STRING \
         | $VAWK --header '\$6>=$MINQUAL && I\$RPR>0 && I\$RPL>0' \
